src/components/counter.js

```diff
@@ -40,21 +34,37 @@ export class Learning_Project extends Component {
   render() {
     return (
       <View style={styles.counter_col}>
-        <View style={styles.counter_row_first}>
-          <Button
-            large
-            title="Increment +"
-            color="green"
-            onPress={this.props.actions.counterIncrement}
-          />
-          <Text style={styles.text}>{this.props.count}</Text>
-          <Button
-            large
-            title="Decrement -"
-            onPress={this.props.actions.counterDecrement}
-          />
-        </View>
-        <View style={styles.counter_row}>
+        <Button
+          large
+          title="Add counter"
+          color="blue"
+          onPress={this.props.actions.newCounter}
+        />
+        {this.props.counterList.map((counter, index) => {
+          return (
+            <View style={styles.counter_row} key={index}>
+              <Button
+                large
+                title="Increment +"
+                color="green"
+                onPress={() => {
+                  this.props.actions.counterIncrement(index);
+                  this.forceUpdate();
+                }}
+              />
+              <Text style={styles.text}>{counter.count}</Text>
+              <Button
+                large
+                title="Decrement -"
+                onPress={() => {
+                  this.props.actions.counterDecrement(index);
+                  this.forceUpdate();
```

Comment:
I'm happy you found a solution to the rerender problem, but this is more of a bandage than an actual solution. Because it depends on counterDecrement being a synchronous action, otherwise when forceUpdate fires the state will still be the same. While forceUpdate does exist it should be avoided unless absolutely needed.